cartridges/int_adyen_overlay/cartridge/js/pages/checkout/adyen-cse.js

```diff
@@ -35,14 +35,19 @@
                         $('#dwfrm_billing_paymentMethods_creditCard_selectedCardID').val("");
                         copyCardData(window.AdyenCard);
                     }
-                    
-                    e.preventDefault();
-                    $('.form-data-error').html('');
-                    $('#billing-submit-hidden').trigger('click');
-
                 }
-            });
+                else if (radioVal == "Adyen"){
+                    var selectedMethod = $('[name="brandCode"]:checked').val();
+                    if(!componentDetailsValid(selectedMethod)){
+                        return false;
+                    }
```

Comment:
don't think this if statement is needed. I believe componentDetailsValid will provide you back true/false. So you can simplify by return componentDetailsValid(selectedMethod)

Reply:
You are right, will update it